### PR TITLE
[FIX] project_timesheet_holidays: wrong public holidays encoded

### PR DESCRIPTION
**Pull request by momegahed**
Steps to reproduce:
1- install  payroll, employees, timesheet, time off 
2- create public holiday ph1 tied to 40h work entry type 
3- create public holiday ph2 tied to 35h work entry type 
4- create a new employee emp with 40h work entry type 
5- click on timesheets
6- ph1 is reflected on his timesheet
7- edit emp to 35h work entry
8- click on timesheets
9- ph1 is still shown instead of ph2

Bug:
no logic handles the change of existing employee work entry type

OPW-3394194

### CODE REVIEW

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left a tiny comment. 🙂 

May I ask you to add a test to reproduce the issue? 

Thanks in advance! 🙂

File: addons/project_timesheet_holidays/models/hr_employee.py [hunk 1]
```diff
@@ -17,6 +17,13 @@ def create(self, vals):
         self._create_future_public_holidays_timesheets(employees)
         return employees
 
+    def _delete_current_public_holidays_timesheets(self):
```
Comment (nle-odoo) [inline addons/project_timesheet_holidays/models/hr_employee.py:20]: detail: method could be positioned at the end (it's not always respected, but in the [guideline](https://www.odoo.com/documentation/16.0/contributing/development/coding_guidelines.html#symbols-and-conventions) the business method are grouped together, separately from CRUD method

**Review by nle-odoo — APPROVED** [review]
looks good to me

**Comment by HANNICHE-Walid** [comment]
https://github.com/odoo/odoo/pull/136914
already fixed

**Pull request closed without merging**